diff --git a/build/esm/sdk/esmLoader.js b/build/esm/sdk/esmLoader.js
index 65f3159c232ba365e7302322b5a4337389c634fa..3f9a21b6a85406417431f2aa5abad6ec331749e2 100644
--- a/build/esm/sdk/esmLoader.js
+++ b/build/esm/sdk/esmLoader.js
@@ -20,7 +20,7 @@ function initializeEsmLoader() {
     try {
       const { addHookMessagePort } = createAddHookMessageChannel();
       // @ts-expect-error register is available in these versions
-      moduleModule.register('import-in-the-middle/hook.mjs', import.meta.url, {
+      moduleModule.register('import-in-the-middle/hook.mjs', new URL('esmLoader.js', import.meta.url), {
         data: { addHookMessagePort, include: [] },
         transferList: [addHookMessagePort],
       });
