diff --git a/out/util/filter.js b/out/util/filter.js
index 369783eff5c64c342831dd3544c3f51e490b0e93..7c62723531bc671ee1ae2cd0be36d5297068170e 100644
--- a/out/util/filter.js
+++ b/out/util/filter.js
@@ -41,7 +41,9 @@ function createFilter(src, patterns, excludePatterns) {
         let relative = getRelativePath(file, srcWithEndSlash, stat);
         // filter the root node_modules, but not a subnode_modules (like /appDir/others/foo/node_modules/blah)
         if (relative === "node_modules") {
-            return false;
+            // return false;
+            // HACK: we want the root node_modules
+            return true;
         }
         else if (relative.endsWith("/node_modules")) {
             relative += "/";
