diff --git a/lib/find_process.js b/lib/find_process.js
index 34188eec28c58340f62c4a0474a3d00c74ed2bde..c8dc18ef98e3a49addcce6989ef62072d5d84a4e 100644
--- a/lib/find_process.js
+++ b/lib/find_process.js
@@ -127,6 +127,9 @@ const finders = {
       proc.stdout.on('data', data => {
         lines.push(data.toString())
       })
+      proc.on('error', err => {
+        reject(new Error('Command \'' + cmd + '\' failed with reason: ' + err.toString()))
+      })
       proc.on('close', code => {
         if (code !== 0) {
           return reject(new Error('Command \'' + cmd + '\' terminated with code: ' + code))
